packages/sources/finnhub/src/transport/market-status.ts

```
@@ -0,0 +1,161 @@
+import { MarketStatus } from '@chainlink/external-adapter-framework/adapter'
+import { HttpTransport } from '@chainlink/external-adapter-framework/transports'
+import { makeLogger } from '@chainlink/external-adapter-framework/util'
+
+import type { BaseEndpointTypes } from '../endpoint/market-status'
+
+export const markets = [
+  'AD',
+  'AS',
+  'AT',
+  'AX',
+  'BA',
+  'BC',
+  'BD',
+  'BE',
+  'BH',
+  'BK',
+  'BO',
+  'BR',
+  'CA',
+  'CN',
+  'CO',
+  'CR',
+  'CS',
+  'DB',
+  'DE',
+  'DU',
+  'F',
+  'HE',
+  'HK',
+  'HM',
+  'IC',
+  'IR',
+  'IS',
+  'JK',
+  'JO',
+  'KL',
+  'KQ',
+  'KS',
+  'KW',
+  'L',
+  'LS',
+  'MC',
+  'ME',
+  'MI',
+  'MT',
+  'MU',
+  'MX',
+  'NE',
+  'NL',
+  'NS',
+  'NZ',
+  'OL',
+  'PA',
+  'PM',
+  'PR',
+  'QA',
+  'RO',
+  'RG',
+  'SA',
+  'SG',
+  'SI',
+  'SN',
+  'SR',
+  'SS',
+  'ST',
+  'SW',
+  'SZ',
+  'T',
+  'TA',
+  'TL',
+  'TO',
+  'TW',
+  'TWO',
+  'US',
+  'V',
+  'VI',
+  'VN',
+  'VS',
+  'WA',
+  'HA',
+  'SX',
+  'TG',
+  'SC',
+] as const
+
+export type Market = (typeof markets)[number]
+
+function isMarket(v: any): v is Market {
+  return markets.includes(v as Market)
+}
+
+// See: https://finnhub.io/docs/api/market-status
+type ResponseBody = {
+  exchange: string // US
+  holiday: string | null // Christmas
+  isOpen: boolean // false
+  session: string | null // pre-market
+  timezone: string // America/New_York
+  t: number // 1697018041
+}
+
+export type HttpEndpointTypes = BaseEndpointTypes & {
+  Provider: {
+    RequestBody: never
+    ResponseBody: ResponseBody
+  }
+}
+
+const logger = makeLogger('FinnhubMarketStatusEndpoint')
+
+export const transport = new HttpTransport<HttpEndpointTypes>({
+  prepareRequests: (params, config) => {
+    return params.map((param) => {
+      const market = param.market
+      if (!isMarket(market)) {
```

I think you could do this check in a `customInputValidation` block in the `endpoint/market-status` file to fail fast (throw an AdapterError).

Example: https://github.com/smartcontractkit/external-adapters-js/blob/main/packages/sources/nav-generic/src/endpoint/nav.ts#L34-L37

Reason you'd want to fail fast is to avoid sending a param to the DP you know will be rejected. And the reason you'd want to do it in the endpoint file vs in this `prepareRequests` block is if you return an error here, it will retry outside of the built-in rate limiter, which can be confusing.

---

No longer necessary due to validation on which markets can be supplied